[Jsjk112344] iP

Jsjk112344 opened the pull request:
### Hirono: Your Personal Task Manager

> "The art of progress is to preserve order amid change and to preserve change amid order." – Alfred North Whitehead 🧠

Hirono is a text-based task management application designed to help you stay organized and productive. It's:

1. Easy to use
2. Efficient
3. Customizable for your needs

**Features**
1. Managing Tasks
2. Add, delete, mark, unmark, and find tasks with intuitive commands.
3. Event Scheduling
4. Keep track of events and deadlines with precise time management.
5. Find Tasks Quickly
6. Search for tasks by keywords or phrases.
7. Effortless Navigation
8. Easily list or filter tasks by date.

**Getting Started 🚀**
Run the application via java -jar hirono.jar.
Add your first task:
Example: todo read book
Explore the features and let Hirono organize your life!

**Task Overview**

- [ ] Implement task creation (ToDo, Deadline, Event)
- [ ]  Add task marking/unmarking
- [ ]  Implement search and filter functions
- [ ] Work on GUI

**Example Usage**

`public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
`

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Jsjk112344 committed:
completed first task, basic skeleton

Jsjk112344 committed:
Added echoing functionality

Jsjk112344 committed:
completed level 2 add lists

Jsjk112344 committed:
Did task 3, can now mark tasks as done

Jsjk112344 committed:
Completed task 4 w inheritance

Jsjk112344 committed:
added tests

Jsjk112344 committed:
Completed task 5 error handling

Jsjk112344 committed:
Completed level 6

Jsjk112344 committed:
Added save functionality

Jsjk112344 committed:
final bug fixes level 7

Jsjk112344 committed:
Merge branch 'branch-Level-7'

Jsjk112344 committed:
updated the deadline task to parse dates

Jsjk112344 committed:
Added functionality to find events by date

Jsjk112344 committed:
Merge branch 'branch-Level-8'

Jsjk112344 committed:
Added Ui Class

Jsjk112344 committed:
Added a Parser Class

Jsjk112344 committed:
added some packages

Jsjk112344 committed:
Added Command class

Jsjk112344 committed:
Merge branch 'branch-More-OOP'

Jsjk112344 committed:
A-Packages

Jsjk112344 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Jsjk112344 committed:
Add Gradle support to manage dependencies and automate builds

Jsjk112344 committed:
Add minimal tests for Parser and TaskList classes

Jsjk112344 committed:
Add Storage tests and more TaskListTests

Jsjk112344 committed:
Merge remote-tracking branch 'origin/Junit-Tests'

Jsjk112344 committed:
Add jar file

Jsjk112344 committed:
Add Javadoc comments to most public classes and methods

Jsjk112344 committed:
Merge remote-tracking branch 'origin/branch-ACodingStandard'

Jsjk112344 committed:
Add Styling for switch case

Jsjk112344 committed:
Merge remote-tracking branch 'origin/branch-A-CodingStandard'

Jsjk112344 committed:
Add FindCommand

Jsjk112344 committed:
Add FindCommand and findcommand method

Jsjk112344 committed:
Complete implementation of Find

Jsjk112344 committed:
Add tests for findevents

Jsjk112344 committed:
syntax fixes

Jsjk112344 committed:
Merge remote-tracking branch 'origin/branch-Level-9'

Jsjk112344 committed:
Fix errors caused by bad merge conflict resolution

Jsjk112344 committed:
add javadoc comments to commands

Jsjk112344 committed:
add checkStyle files and fix style issues

Jsjk112344 committed:
Merge remote-tracking branch 'origin/branch-A-CheckStyle'

zuoshihua commented on src/main/java/hirono/Ui.java:
```diff
@@ -0,0 +1,94 @@
+     * Displays the welcome message and the logo for the application.
+     */
+    public void showWelcome() {
+        String logo =
+                  "                            .-'''-.                   .-'''-.     \n"
+                + "                           '   _    \\                '   _    \\   \n"
+                + "   .        .--.         /   /` '.   \\    _..._    /   /` '.   \\  \n"
+                + " .'|        |__|        .   |     \\  '  .'     '. .   |     \\  '  \n"
+                + "<  |        .--..-,.--. |   '      |  '.   .-.   .|   '      |  ' \n"
+                + " | |        |  ||  .-. |\\    \\     / / |  '   '  |\\    \\     / /  \n"
+                + " | | .'''-. |  || |  | | `.   ` ..' /  |  |   |  | `.   ` ..' /   \n"
+                + " | |/.'''. \\|  || |  | |    '-...-'`   |  |   |  |    '-...-'`    \n"
+                + " |  /    | ||  || |  '-                |  |   |  |                \n"
+                + " | |     | ||__|| |                    |  |   |  |                \n"
+                + " | |     | |    | |                    |  |   |  |                \n"
+                + " | '.    | '.   |_|                    |  |   |  |                \n"
+                + " '---'   '---'                         '--'   '--'                \n";
```
Love the banner. Good job!

zuoshihua commented on src/main/java/hirono/Storage.java:
```diff
@@ -0,0 +1,177 @@
+        TaskList taskList = new TaskList();
+        File file = new File(filePath);
+
+        // Ensure the file exists
+        if (!file.exists()) {
+            File parentDir = file.getParentFile();
+            if (parentDir != null) {
+                parentDir.mkdirs();
+            }
+            file.createNewFile();
+        }
```
This seems very similar to `ensureFileExists()`. Perhaps it can be called here instead?

zuoshihua commented on src/main/java/hirono/Storage.java:
```diff
@@ -0,0 +1,177 @@
+                String description = parts[2].trim();
+
+                switch (type) {
+                case "T":
+                    ToDo todo = new ToDo("todo " + description);
+                    if (isDone) {
+                        todo.markAsDone();
+                    }
+                    taskList.addLoadedTask(todo);
+                    break;
+
+                case "D":
+                    Deadline deadline = new Deadline(
+                        "deadline " + description + " /by " + parts[3].trim()
+                    );
+                    if (isDone) {
+                        deadline.markAsDone();
+                    }
+                    taskList.addLoadedTask(deadline);
+                    break;
+
+                case "E":
+                    Event event = new Event(
+                        "event " + description + " /from " + parts[3].trim() + " /to " + parts[4].trim()
+                    );
+                    if (isDone) {
+                        event.markAsDone();
+                    }
+                    taskList.addLoadedTask(event);
+                    break;
+
+                default:
+                    throw new HironoException("Invalid task type in file.");
```
It seems `Storage` is providing functionality for parsing all types of tasks. Perhaps each type of task can provide its own parsing functionality to offload this burden on `Storage`.

zuoshihua commented on src/main/java/hirono/tasks/ToDo.java:
```diff
@@ -0,0 +1,34 @@
+/**
+ * Describes a basic task to do
+ */
+public class ToDo extends Task {
```
Perhaps rename to `Todo`?

zuoshihua commented on src/main/java/hirono/Hirono.java:
```diff
@@ -0,0 +1,64 @@
+                if (command.isExit()) {
+                    break;
+                }
+            } catch (HironoException e) {
+                // Show error message for invalid commands or execution issues
+                ui.showError(e.getMessage());
+            }
```
`execute()` may throw `IOException`. Perhaps add another catch block to handle it.

zuoshihua reviewed (commented):
LGTM! Good effort on maintaining coding standards.

Jsjk112344 committed:
Add JavaFX  configuration

Jsjk112344 committed:
Fix style issues

Jsjk112344 committed:
small syntax changes

Jsjk112344 committed:
Merge remote-tracking branch 'origin/branch-Level-10'

Jsjk112344 committed:
edit build.gradle to create jar file

Jsjk112344 committed:
Add Assertions to Critical Methods
Adding assertions in critical methods across the project to ensures key invariants and assumptions hold during runtime.

Adding assertions validates:

Successful addition of tasks to the task list.
Proper removal of tasks from the task list.
Valid format and presence of search terms in the findTasks method.
Valid date formats in the getEventsOnDate method.
These improve the robustness of the code by catching logic errors during development. Assertions ensure assumptions about data states and inputs are met, preventing unexpected behavior and simplifying debugging.

lyhthaddeus commented on data/hirono.txt:
```diff
@@ -0,0 +1,6 @@
+D | 0 | return book | 2023-06-11 1800
```
is pushing the data onto github needed?

lyhthaddeus commented on src/main/java/data/hirono.txt:
```diff
@@ -0,0 +1,3 @@
+T | 0 | read book
```
is there two instance of your data?

lyhthaddeus commented on src/main/java/hirono/HironoException.java:
```diff
@@ -0,0 +1,18 @@
+ * This exception is thrown when an error occurs during user input processing
+ * or command execution within the application.
+ */
+public class HironoException extends Exception {
```
could you make a package for the exceptions?

lyhthaddeus commented on src/main/java/hirono/Hirono.java:
```diff
@@ -0,0 +1,104 @@
+     * @throws IOException     If there is an error reading from or writing to the storage file.
+     * @throws HironoException If an error occurs during command execution.
+     */
+    public void run() throws IOException, HironoException {
```
is there a need to throw exceptions if you already catching them?

lyhthaddeus commented on src/main/java/hirono/parser/Parser.java:
```diff
@@ -0,0 +1,62 @@
+     * @throws HironoException
+     */
+    public Command parse(String input) throws HironoException {
+        if (input.equals("list")) {
```
could a switch case using enums be an alternative?

lyhthaddeus commented on src/main/java/hirono/task/Deadline.java:
```diff
@@ -0,0 +1,133 @@
+     * @return {@code true} if the description matches the deadline format, {@code false} otherwise.
+     */
+    private boolean isValidDeadline(String description) {
+        String deadlineRegex = "^deadline\\s+.+\\s+/by\\s+.+$";
```
I love the use of regex here to parse the text input :)

lyhthaddeus reviewed (commented):
Overall nice code. really good use of OOP concepts.
Code was easy to read and follow
I love the use of regex to parse data.

Jsjk112344 committed:
Add streams and move logic from tasklist to individual command classes

Jsjk112344 committed:
move logic for deleteTask and unMark task to their respective classes instead of in tasklist

Jsjk112344 committed:
Refactor TaskList methods to use Java Streams

Refactor the findTasks and getEventsOnDate methods in the TaskList class to leverage the Java Streams API for better readability and efficiency.

Replace traditional loops with declarative Stream-based operations for filtering and mapping tasks.
Simplify logic for finding matching tasks by directly using filter, map, and toList methods.
Reduce boilerplate code in getEventsOnDate by filtering tasks using a single Stream pipeline.

Jsjk112344 committed:
fix code quality issues

Jsjk112344 committed:
Merge pull request #1 from Jsjk112344/branch-A-Assertions

A-Assertions

Jsjk112344 committed:
Merge pull request #2 from Jsjk112344/branch-A-CodeQuality

fix code quality issues

Jsjk112344 committed:
Merge branch 'master' into branch-A-Streams

Jsjk112344 committed:
Merge pull request #3 from Jsjk112344/branch-A-Streams

Branch a streams

Jsjk112344 committed:
fix style errors

Jsjk112344 committed:
Set up GitHub Actions CI with Java

Jsjk112344 committed:
Merge remote-tracking branch 'origin/branch-A-CI'

Jsjk112344 committed:
edits to timezone

Jsjk112344 committed:
Add time logic

Jsjk112344 committed:
Add fixes to parsing dates

Jsjk112344 committed:
fix style issues

Jsjk112344 committed:
Add task editing functionality with type preservation

Add new EditCommand to allow users to modify existing tasks while preserving their types.
This implementation includes robust error handling and maintains task status after edits.

Key Features:
- Edit task descriptions while preserving original task type (todo/deadline/event)
- Maintain task completion status after editing
- Support for all task types with their specific formats:
  * todo tasks
  * deadline tasks with dates
  * event tasks with from/to times

Jsjk112344 committed:
Merge remote-tracking branch 'origin/branch-BCDExtension'

Jsjk112344 committed:
Add more styling and error message highlighting

Jsjk112344 committed:
Add resizing functionality

Jsjk112344 committed:
Merge remote-tracking branch 'origin/branch-A-BetterGUI'

Jsjk112344 committed:
Add website content for Hirono

Jsjk112344 committed:
add image and changes to color styling

Jsjk112344 committed:
update image

Jsjk112344 committed:
update website

Jsjk112344 committed:
Update image name

Jsjk112344 committed:
Update README.md

Jsjk112344 committed:
Update README.md

Jsjk112344 committed:
Update README.md

Jsjk112344 committed:
update readme

Jsjk112344 committed:
Merge branch 'master' of https://github.com/Jsjk112344/ip

Jsjk112344 committed:
Update README.md

Jsjk112344 committed:
update image name

Jsjk112344 committed:
Edit readme

Jsjk112344 committed:
Update readme format

Jsjk112344 committed:
Fix style errors and update README with usage documentation

Fix style inconsistencies throughout the codebase and expand README
with comprehensive usage documentation for new users.

Style fixes:
- Fix inconsistent indentation in source files
- Standardize line endings across files
- Remove trailing whitespace
- Correct brace placement per style guide

README improvements:
- Add detailed usage instructions with examples
- Include command reference section
- Document system requirements and date/time formats

These changes improve code maintainability and make the project more
accessible to new contributors.
